cfme/storage/volume_snapshot.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+import attr
+
+from navmazing import NavigateToSibling, NavigateToAttribute
+from widgetastic_manageiq import (
+    Accordion,
+    BaseEntitiesView,
+    BreadCrumb,
+    ItemsToolBarViewSelector,
+    ManageIQTree,
+    SummaryTable
+)
+from widgetastic_patternfly import Button, Dropdown, FlashMessages
+from widgetastic.widget import View, Text
+
+from cfme.base.ui import BaseLoggedInPage
+from cfme.common import TagPageView, WidgetasticTaggable
+from cfme.exceptions import SnapshotNotFoundError, ItemNotFound
+from cfme.modeling.base import BaseCollection, BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from cfme.utils.wait import wait_for
+
+
+class VolumeSnapshotToolbar(View):
+    """The toolbar on the Volume Snapshot page"""
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class VolumeSnapshotDetailsToolbar(View):
+    """The toolbar on the Volume Snapshot detail page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button('Download summary in PDF format')
+
+
+class VolumeSnapshotDetailsEntities(View):
+    """The entities on the Volume Snapshot detail page"""
+    breadcrumb = BreadCrumb()
+    properties = SummaryTable('Properties')
+    relationships = SummaryTable('Relationships')
+    smart_management = SummaryTable('Smart Management')
+
+
+class VolumeSnapshotDetailSidebar(View):
+    """The accordion on the Volume Snapshot details page"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class VolumeSnapshotView(BaseLoggedInPage):
+    """A base view for all the Volume Snapshot pages"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]')
+
+    @property
+    def in_volume_snapshot(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Storage', 'Block Storage', 'Volume Snapshots']
+        )
+
+    @property
+    def is_displayed(self):
+        return self.in_volume_snapshot
+
+
+class VolumeSnapshotAllView(VolumeSnapshotView):
+    """The all Volume Snapshot page"""
+    toolbar = View.nested(VolumeSnapshotToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_volume_snapshot and
+            self.title.text == 'Cloud Volume Snapshots')
```

Comment:
**REQUIRED**: If this is listing snapshots for a particular volume, then the is_displayed should check also the name of the volume from the UI. In case that is not possible (which would qualify for a bug against CFME anyway I guess), then return false.

Reply:
@mfalesni this is_displayed method for normal All page [Storage>Block Storage> Volume Snapshots] showing all available snapshots for different volumes. Not for a particular volume.